src/types/pick.test.ts

```diff
@@ -0,0 +1,219 @@
+    union: UnionTestType,
+    intersection: pick(IntersectionTestType, ['a', 'd']),
+    inlineUnion: literal('stringliteral').or(literal(12)),
+    nestedOptionals: object({
+        req: string,
+    }).withOptional({
+        opt: string,
+    }),
+    arr: array(number).withConstraint('nonempty', i => i.length > 0),
+}).withOptional({
+    optionalUnion: pick(UnionTestType, ['a']).or(pick(IntersectionTestType, ['c', 'd'])),
+    optionalIntersection: pick(IntersectionTestType, ['a', 'd']),
+});
+testTypeImpl({
+    type: pick(NastyType, ['inlineUnion', 'optionalIntersection']),
+    name: `Pick<NastyType, 'inlineUnion' | 'optionalIntersection'>`,
+    basicType: 'object',
+    validValues: [
+        { inlineUnion: 'stringliteral', optionalIntersection: { a: 'reqstring', d: 'optionalstring' } },
+        { inlineUnion: 12, optionalIntersection: { a: 'reqstring' } },
+        { inlineUnion: 'stringliteral' },
+    ],
+    invalidValues: [
+        [
+            { inlineUnion: 13, optionalIntersection: { a: 'reqstring', d: 'optionalstring' } },
+            [
+                `error in [Pick<NastyType, 'inlineUnion' | 'optionalIntersection'>] at <inlineUnion>: in union element [12]: expected the literal 12, got: 13`,
+                `  • disregarded 1 union-subtype that does not accept a number`,
+            ],
+        ],
+        [
+            { optionalIntersection: { a: 'reqstring', d: 'optionalstring' } },
+            `error in [Pick<NastyType, 'inlineUnion' | 'optionalIntersection'>]: missing property <inlineUnion> ["stringliteral" | 12], got: { optionalIntersection: { a: "reqstring", d: " .. " } }`,
+        ],
+        [
+            { inlineUnion: 12, optionalIntersection: { a: 'reqstring', d: undefined } },
+            `error in [Pick<NastyType, 'inlineUnion' | 'optionalIntersection'>] at <optionalIntersection.d>: expected a string, got an undefined`,
+        ],
+        [
+            { inlineUnion: 12, optionalIntersection: {} },
+            `error in [Pick<NastyType, 'inlineUnion' | 'optionalIntersection'>] at <optionalIntersection>: missing property <a> [string], got: {}`,
+        ],
+    ],
+});
+
+const example1 = createExample(NastyType, 1);
+const example2 = createExample(NastyType, 2);
+
+testTypeImpl({
+    type: pick(NastyType, ['optionalUnion', 'arr']),
+    name: `Pick<NastyType, 'optionalUnion' | 'arr'>`,
+    basicType: 'object',
+    validValues: [
+        { optionalUnion: { a: 'pickedUnion' }, arr: [1] },
+        { optionalUnion: { c: 12, d: 'pickedIntersection' }, arr: [12, 3] },
+        { optionalUnion: { c: 1 }, arr: [12, 3] },
+        { arr: [12, 3] },
+    ],
+    invalidValues: [
+        [
+            { optionalUnion: 13, arr: [] },
+            [
+                `errors in [Pick<NastyType, 'optionalUnion' | 'arr'>]:`,
+                `- at <optionalUnion>: expected an object, got a number (13)`,
+                `- at <arr>: expected a [nonempty], got: []`,
```

Comment:
Hint, this fails on newlines, but those are presented poorly in the Jest output:
![image](https://github.com/skunkteam/types/assets/5168053/3b6594d0-4a64-4c6c-85ce-b5280db73e67)


```suggestion
                `errors in [Pick<NastyType, 'optionalUnion' | 'arr'>]:`,
                '',
                `- at <optionalUnion>: expected an object, got a number (13)`,
                '',
                `- at <arr>: expected a [nonempty], got: []`,
```